cdci_data_analysis/analysis/queries.py

```diff
@@ -375,11 +375,9 @@ def test_communication(self, instrument, query_type='Real', logger=None, config=
             query_out.set_done(message=message, debug_message=str(debug_message),status=status)
 
         except ConnectionError as e:
```

Comment:
I am most uncertain about this `ConnectionError`. I guess it never raises. Here, only `InternalError` may happen: the backend is either reachable or not. As of now, it may be caught as `Exception`, since they all are to be treated similarly: sentry and advise to the user to retry and we are working.

Reply:
A bit dangerous to get rid without a test! But yeah, this only happens in exceptional cases. So a change here would not break too much. But still, would be nice to have a test.

Reply:
It may be a lot to do right now. But maybe not. Could you try? So it will fit in the next patch, it should not be more than couple of days. 
If too long, a more minimal approach would suffice.

Reply:
So, the goal of the test is to verify that `ConnectionError` is never raised? Is every plugin implementing the `test_communication` differently?

Reply:
There is no guarantee that what is raised is `ConnectionError`. Certainly it never happens with INTEGRAL plugin. Anyway, no matter what is raised there it is an internal error. Is there really any other possibility in testing communication? There can not be any bad request, and the result can not be unusual but acceptable.